samples/sample_decode/src/pipeline_decode.cpp

```diff
@@ -647,6 +647,9 @@ mfxStatus CDecodingPipeline::InitMfxParams(sInputParams *pParams)
         }
         if (MFX_ERR_MORE_DATA == sts)
         {
+            if (m_mfxBS.DataFlag & MFX_BITSTREAM_EOS)
+                break;
```

Comment:
I think we should put this ```break``` on line 664 instead

Reply:
is it possible that m_FileReader->ReadNextFrame() returns MFX_ERR_MORE_DATA but m_mfxBS has valid data?

Reply:
Hm, I looked at the code and currently I don't understand how it will fall into forever loop?
In my undenderstanding the workflow is following
1) ```DecodeHeader``` return ```ERR_MORE_DATA```
2) ```ReadNextFrame``` will return ```ERR_MORE_DATA``` again
3) ```EOS``` set
4) ```continue```
5) repeat 1)
6) repeat 2)
7) as ```EOS``` already set, return error

Reply:
EOS is cleared in `ReadNextFrame`.

Reply:
ok, now I see. the problem is that each overload of ```ReadNextFrame``` rewrites state of ```pBS->DataFlag```. In my understanding the correct fix would be to introduce early exit in these function if they detect ```EOS``` in input stream. But such change may affect our validation, so it should go through some smoke val first. I'm going to treat your fix as WA for such problem and I will remove my -1. Also if you are in hurry with this fix, I'm ok to merge it but with TODO comment in code that this is WA and description of proper fix. Other ways I would prefer to fix it in ```ReadNextFrame```, but that will take some time

Reply:
I agree my fix is a workaround for the issue and the better fix is to introduce early exit in each `ReadNextFrame`, however what I need is a workable tool and the WA is simpler. I added some comment about the WA but I'm fine if you won't merge this WA.